File: test/inheritance.java
// package test;

class Base {
    int x;

    Base(int a) {
        x = a;
        System.out.println("Base Class Constructor: x = " + x);
    }
}

class Derived extends Base {
    int y;

    Derived(int a, int b) {
        super(a);
        y = b;
        System.out.println("Derived Class Constructor: y = " + y);
    }
}

public class inheritance {
    public static void main(String[] args) {
        Derived ob = new Derived(10, 20);
        System.out.println(ob.getClass());
    }
}

File: test/inheritance.cpp
#include <iostream>
using namespace std;

class Base
{
public:
    int x;

    Base(int a) : x(a)
    {
        cout << "Base Class Constructor: x = " << x << endl;
    }

    ~Base()
    {
        cout << "Base class Destructor" << endl;
    }
};

class Derived : public Base
{
public:
    int y;

    Derived(int a, int b) : Base(a)
    {
        y = b;
        cout << "Derived Class Constructor: y = " << y << endl;
    }

    ~Derived()
    {
        cout << "Derived class Destructor" << endl;
    }
};


int main() {
    Derived ob(5, 6);
    return 0;
}